### PR TITLE
Fplib build fix

### PR DESCRIPTION
**Pull request by Jeremy-Kemp**
Related to #25 and #604.

The change re-names fplib.c to fplib.cpp and also changes the conditions under which it is built (to match when it is invoked by conversions).

### CODE REVIEW

**Comment by Jeremy-Kemp** [comment]
Yes I think having some sort of arch variable that we define in the top level `CMakeLists.txt` will make any dependant build decisions a lot easier to maintain.

I'm looking forward to `travis-arm`, it'll be a great step forward for CTS :).

**Comment by kpet** [comment]
I've got the Arm and AArch64 CI ready (and failing because of this issue) now. I'll rebase the change and create a PR once we get a fix for this issue in.

**Comment by kpet** [comment]
@Jeremy-Kemp Do you want to add the arch variable as part of this PR or get this in as is and do it afterwards?

**Comment by Jeremy-Kemp** [comment]
> @Jeremy-Kemp Do you want to add the arch variable as part of this PR or get this in as is and do it afterwards?

I can do this now. I'll cobble something together and update this review.

File: CMakeLists.txt [hunk 1]
```diff
@@ -121,14 +121,26 @@ endif (GL_IS_SUPPORTED AND CLConform_GL_LIBRARIES_DIR)
 include(CheckFunctionExists)
 include(CheckIncludeFiles)
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm.*|ARM.*)" OR CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
+    set(CLConform_TARGET_ARCH ARM)
```
Comment (kpet) [inline CMakeLists.txt]: I suggest we use two different values for `ARM` and `ARM64`.

File: test_conformance/conversions/CMakeLists.txt [hunk 2]
```diff
@@ -13,9 +13,9 @@ set (${MODULE_NAME}_SOURCES
       Sleep.cpp test_conversions.cpp basic_test_conversions.cpp
 )
 
-if(ANDROID)
-    list(APPEND CONVERSIONS_SOURCES fplib.c)
-endif(ANDROID)
+if(CMAKE_COMPILER_IS_GNUCC AND (${CLConform_TARGET_ARCH} EQUAL "ARM" OR ${CLConform_TARGET_ARCH} EQUAL "ARM64"))
```
Comment (kpet) [inline test_conformance/conversions/CMakeLists.txt]: This file is actually not required for ARM64.

**Pull request merged**